Code/Source/RobotImporter/URDF/CollidersMaker.cpp

```diff
@@ -0,0 +1,309 @@
+                azMeshPath.c_str(),
+                assetInfo,
+                watchDir);
+
+            // Set export method to convex mesh
+            auto readOutcome = AZ::JsonSerializationUtils::ReadJsonFile(assetInfoFilePath.c_str());
+            if (!readOutcome.IsSuccess())
+            {
+                AZ_Error("CollisionMaker", false, "Could not read %s", assetInfoFilePath.c_str());
+                return;
+            }
+            rapidjson::Document assetInfoJson = readOutcome.TakeValue();
+            auto manifestObject = assetInfoJson.GetObject();
+            auto valuesIterator = manifestObject.FindMember("values");
+            if (valuesIterator == manifestObject.MemberEnd())
+            {
+                AZ_Error("CollisionMaker", false, "Invalid json file: %s", assetInfoFilePath.c_str());
+                return;
+            }
+            auto valuesArray = valuesIterator->value.GetArray();
+            for (auto& value : valuesArray)
+            {
+                auto object = value.GetObject();
+
+                auto physXMeshGroupIterator = object.FindMember("$type");
+                AZStd::basic_string<char> physXMeshGroupType = "{5B03C8E6-8CEE-4DA0-A7FA-CD88689DD45B} MeshGroup";
+
+                if (AZ::StringFunc::Equal(physXMeshGroupIterator->value.GetString(), physXMeshGroupType))
+                {
+                    value.AddMember(rapidjson::StringRef("export method"), rapidjson::StringRef("1"), assetInfoJson.GetAllocator());
+                }
+            }
+
+            auto saveOutcome = AZ::JsonSerializationUtils::WriteJsonFile(assetInfoJson, assetInfoFilePath.c_str());
+            if (!saveOutcome.IsSuccess())
+            {
+                AZ_Error("CollisionMaker", false, "Could not save %s", assetInfoFilePath.c_str());
+                return;
+            }
+
+            // Add asset to expected assets list
+            if (assetFound)
+            {
+                auto pxmeshAssetPath = AZStd::string(assetInfo.m_relativePath.c_str());
+                AzFramework::StringFunc::Path::ReplaceExtension(pxmeshAssetPath, ".pxmesh");
+
+                AZStd::lock_guard lock{ m_buildMutex };
+                m_meshesToBuild.push_back(AZStd::string(pxmeshAssetPath.c_str()));
+            }
+        }
+    }
+
+    void CollidersMaker::AddColliders(urdf::LinkSharedPtr link, AZ::EntityId entityId)
+    {
+        AZStd::string typeString = "collider";
+        size_t nameSuffixIndex = 0; // For disambiguation when multiple unnamed colliders are present. The order does not matter here
+        for (auto collider : link->collision_array)
+        { // one or more colliders - the array is used
+            AddCollider(collider, entityId, PrefabMakerUtils::MakeEntityName(link->name.c_str(), typeString, nameSuffixIndex));
+            nameSuffixIndex++;
+        }
+
+        if (link->collision_array.size() == 0)
```

Comment:
repeat: `.empty()` can be used to better clarify intent.
```suggestion
        if (link->collision_array.empty())
```